### PR TITLE
refactor(pm_list): Update required fields for a payment method

### PR DESCRIPTION
**Pull request by prasunna09**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Refactored  required fields for a payment method and removed fields not collected by sdk such as description, browser info.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by ArjunKarthik — DISMISSED** [review]

File: crates/api_models/src/enums.rs [hunk 1]
```diff
@@ -394,11 +394,12 @@ pub enum FieldType {
     UserAddressCity,
     UserAddressPincode,
     UserAddressState,
-    UserAddressCountry,
+    UserAddressCountry { options: Vec<String> },
```
Comment (Narayanbhat166) [inline crates/api_models/src/enums.rs:397]: Why is this field added? I see that `UserCountry` is also present, if `UserCountry` is not needed remove that enum variant

File: crates/router/src/configs/defaults.rs [hunk 2]
```diff
@@ -646,8 +838,8 @@ impl Default for super::settings::RequiredFields {
                                         RequiredFieldInfo {
                                             required_field: "billing.address.country".to_string(),
                                             display_name: "country".to_string(),
-                                            field_type: enums::FieldType::UserCountry {
-                                                options: vec!["US".to_string(), "IN".to_string()],
+                                            field_type: enums::FieldType::UserAddressCountry {
+                                                options: vec![],
```
[the remark this answers is not in the document]
Comment (prasunna09) [reply crates/router/src/configs/defaults.rs:842]: Here, it is left empty because SDK will take care of countries list if not mentioned. (Discussed with SDK team)

File: crates/api_models/src/enums.rs (hunk 1 above)
Comment (prasunna09) [reply crates/api_models/src/enums.rs:397]: So there is country inside the payment_method_data which is UserCountry, and for country inside Address struct is UserAddressCountry

File: crates/router/src/configs/defaults.rs [hunk 3]
```diff
@@ -1047,7 +1138,7 @@ impl Default for super::settings::RequiredFields {
                             (
                                 enums::Connector::Adyen,
                                 vec![RequiredFieldInfo {
-                                    required_field: "card_holder_name".to_string(),
+                                    required_field: "payment_method_data.bank_debit.ach_bank_debit.card_holder_name".to_string(),
```
Comment (Narayanbhat166) [inline crates/router/src/configs/defaults.rs:1141]: This needs to be generalized, cannot keep using `card_holder_name` for all the payment methods which are not related to cards.

**Review by Narayanbhat166 — APPROVED** [review]

**Review by Chethan-rao — DISMISSED** [review]

File: crates/router/src/configs/defaults.rs [hunk 4]
```diff
@@ -437,48 +413,302 @@ impl Default for super::settings::RequiredFields {
...
+                                vec![
+                                    RequiredFieldInfo {
+                                        required_field: "billing.address.first_name".to_string(),
+                                        display_name: "first_name".to_string(),
+                                        field_type: enums::FieldType::UserBillingName,
+                                    },
+                                    RequiredFieldInfo {
+                                        required_field: "billing.address.line1".to_string(),
+                                        display_name: "line1".to_string(),
+                                        field_type: enums::FieldType::UserAddressline1,
```
Comment (SanchithHegde) [inline crates/router/src/configs/defaults.rs:666]: We might have to consider renaming this and `UserAddressline2` to `UserAddressLine1` and `UserAddressLine2` instead. This can be taken up in a separate PR.

File: crates/router/src/configs/defaults.rs [hunk 5]
```diff
@@ -509,26 +739,21 @@ impl Default for super::settings::RequiredFields {
...
-                                        display_name: "browser_info".to_string(),
-                                        field_type: enums::FieldType::Text,
-                                    },
                                 ],
                             ),
                             (
                                 enums::Connector::Worldline,
                                 vec![RequiredFieldInfo {
-                                    required_field: "card.card_holder_name".to_string(),
+                                    required_field: "payment_method_datacard.card_holder_name".to_string(),
```
Comment (SanchithHegde) [inline crates/router/src/configs/defaults.rs:756]: Was this supposed to be:

```suggestion
                                    required_field: "payment_method_data.card.card_holder_name".to_string(),
```

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/router/src/configs/defaults.rs (hunk 4 above)
Comment (prasunna09) [reply crates/router/src/configs/defaults.rs:666]: So this naming was given by sdk, so did the same, will ask them and modify them in next pr

File: crates/router/src/configs/defaults.rs (hunk 5 above)
Comment (prasunna09) [reply crates/router/src/configs/defaults.rs:756]: yes, missed this, will update now

**Review by SanchithHegde — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]
